smart_card_connector_app/src/window-devices-displaying.js

```diff
@@ -126,6 +129,31 @@ function makeReaderNameForDisplaying(readerName) {
   return readerName;
 }
 
+/**
+ * @param {!GSC.PcscLiteServer.ReaderInfo} reader 
+ * @return {string}
+ */
+function getTooltipText(reader) {
+  let toolTipText;
+  switch (reader['status']) {
+    case GSC.PcscLiteServer.ReaderStatus.INIT:
+      toolTipText = 'initReaderTooltip';
+      break;
+    case GSC.PcscLiteServer.ReaderStatus.SUCCESS:
+      toolTipText = 'successReaderTooltip';
+      break;
+    case GSC.PcscLiteServer.ReaderStatus.FAILURE:
+      toolTipText = 'failureReaderTooltip';
+      break;
+    default:
+      toolTipText = '';
+  }
+  if (reader['isCardPresent']) {
+    toolTipText = 'successCardTooltip';
```

Comment:
nit: This can be moved into the "SUCCESS" case above, making it easier to follow the logic.